### PR TITLE
LibGLSL: Flesh out the compiler a bit.

### PR DESCRIPTION
**Pull request by qeeg**
This will be marked as a draft for the time being as it's nowhere near done yet, but I want to get some opinions on the direction I'm going with this.

### CODE REVIEW

File: Userland/Libraries/LibGLSL/Lexer.cpp [hunk 1]
```diff
@@ -0,0 +1,668 @@
+/*
+ * Copyright (c) 2020, Stephan Unverwerth <s.unverwerth@serenityos.org>
```
Comment (Beyley) [inline Userland/Libraries/LibGLSL/Lexer.cpp:2]: Copyright should prob be updated

Comment (qeeg) [reply Userland/Libraries/LibGLSL/Lexer.cpp:2]: Done.

File: Userland/Libraries/LibGLSL/Lexer.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2020, Stephan Unverwerth <s.unverwerth@serenityos.org>
```
Comment (ADKaster) [inline Userland/Libraries/LibGLSL/Lexer.h:2]: Did stephan write this file?

File: Userland/Libraries/LibGLSL/Lexer.h [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "Token.h"
+
+#include <AK/DeprecatedString.h>
```
Comment (ADKaster) [inline Userland/Libraries/LibGLSL/Lexer.h:11]: It would be preferable to not use DeprecatedString in entirely new code.

**Review by ADKaster — COMMENTED** [review]
This is a decent start I'd say (or continuation?) but it should get some integration into the other parts of LibGLSL/LibGL or at least some unit tests in the same PR before i'd consider it 'ready'

File: Userland/Libraries/LibGLSL/Lexer.h (hunk 2 above)
Comment (qeeg) [reply Userland/Libraries/LibGLSL/Lexer.h:2]: He wrote the LibJS Lexer.h which this is based on.

File: Userland/Libraries/LibGLSL/Lexer.h (hunk 3 above)
Comment (qeeg) [reply Userland/Libraries/LibGLSL/Lexer.h:11]: Again, I based this on the existing LibJS code.

**Comment by qeeg** [comment]
Alright, I'll try converting this code from DeprecatedString to String, but that's gonna require some core AK changes, sorry! ^^

**Comment by kleinesfilmroellchen** [comment]
@qeeg you can make the AK changes separate PRs so we can merge them as soon as possible, as probably many others also need these advancements :^)

**Comment by Poseydon42** [comment]
@qeeg are you still working on this? I'd like to work on the GLSL compiler too, so perhaps we can cooperate somehow so that we don't interfere with each other?

**Comment by qeeg** [comment]
@Poseydon42 Nope, go right ahead.

**Pull request closed without merging**